File: Week3_Intro_HashTables/contains-duplicate-eff.java
import java.util.*;

class Solution {
    public boolean containsDuplicate(int[] nums) {
        /* Initialize set of seen elements */
        Set<Integer> seen = new HashSet<>();

        for (int i : nums) {
            /* If number spotted in seen set, return True */
            if (seen.contains(i))
                return true;
            /* Else, add to seen set */
            seen.add(i);
        }

        /* No duplicates found at this point */
        return false;
    }
}

File: Session1_Intro_HashTables/contains-duplicate-eff.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        # Iniitalize set of seen elements
        seen = set()
        for num in nums:
            # If number is spotted in seen set,
            # Then there is a duplicate
            if num in seen:
                return True
            # Else, add to seen set
            seen.add(num)
        
        # No duplicates are found at this point
        return False
        